qpicard/src/org/qcmg/picard/util/BAMFileUtils.java

```diff
@@ -46,4 +52,47 @@ public static List<String> getContigsFromHeader(SAMFileHeader header) {
 				.collect(Collectors.toList());
 	}
 
+	/**
+	 * picard index name created by replacing .bam to .bai. But we want to have index name end with .bam.bai
+	 * @param bamFile
+	 * @throws Exception 
+	 */	
+	public static void renameBamIndex(File output) throws IOException{
+		String path = output.getPath();
+		
+		//do nothing if not BAM file
+		if( !path.endsWith(FileExtensions.BAM)) return;
+	
+		final String indexFileBase = path.substring(0, path.lastIndexOf('.'));	
+			
+		//rename
+	    final Path indexpicard = IOUtil.addExtension(Paths.get(indexFileBase), FileExtensions.BAI_INDEX);
+	    final Path indexqcmg =  IOUtil.addExtension(Paths.get(path), FileExtensions.BAI_INDEX);
+		 			
+		//rename files
+		Files.move(indexpicard, indexqcmg, StandardCopyOption.REPLACE_EXISTING);				
+	}
+	/**
+	 * picard index name created by adding .bai to .cram. But we want to have index name end with .cram.crai
+	 * @param cramFile
+	 * @throws Exception 
+	 */	
+	
+	public static void renameCramIndex(File output) throws IOException{
```

Comment:
It's usually a good idea to check the input for `public static` methods. What will happen here if output is null?